### PR TITLE
Alg palindrome

### PR DESCRIPTION
**Pull request by sureshrp**

### CODE REVIEW

File: alg-excer1-palindrome.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    if (str[i] !== str[strLength-1-i]) {
+      return input + "- is not Palindrome";
+    }
+  }
+  return input + "- is Palindrome";
+}
+
+console.log(palindrome("SA as"));
+
+console.log(palindrome("345543"));
```
Comment (koding-101) [inline alg-excer1-palindrome.js]: create a function to test your results.

you need to think of more edge cases.

File: alg-excer1-palindrome.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+function palindrome(input) {
+  str =  input.toString().toLowerCase();
+  var strLength= str.length;
+
+  for (var i=0; i<strLength/2; i++) {
```
Comment (koding-101) [inline alg-excer1-palindrome.js]: spacing

File: alg-excer1-palindrome.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+function palindrome(input) {
+  str =  input.toString().toLowerCase();
+  var strLength= str.length;
```
Comment (koding-101) [inline alg-excer1-palindrome.js]: spacing

File: alg-excer1-palindrome.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+  for (var i=0; i<strLength/2; i++) {
+    if (str[i] !== str[strLength-1-i]) {
+      return input + "- is not Palindrome";
+    }
+  }
+  return input + "- is Palindrome";
```
Comment (koding-101) [inline alg-excer1-palindrome.js]: stop adding text to your results!!!! how do you expect to reuse this function if you keep adding text to it? What should it return? What makes more sense?

File: alg-excer1-palindrome.js (hunk 1 above)
Comment (koding-101) [inline alg-excer1-palindrome.js]: what is a palindrome? Do numbers make sense?

File: alg-excer1-palindrome.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+function palindrome(input) {
+  str =  input.toString().toLowerCase();
```
Comment (koding-101) [inline alg-excer1-palindrome.js]: what's missing?

**Comment by koding-101** [comment]
please address basics before i can go over your code

File: alg-excer1-palindrome.js (hunk 1 above)
Comment (sureshrp) [reply alg-excer1-palindrome.js]: updated

File: alg-excer1-palindrome.js (hunk 2 above)
Comment (sureshrp) [reply alg-excer1-palindrome.js]: updated

File: alg-excer1-palindrome.js (hunk 3 above)
Comment (sureshrp) [reply alg-excer1-palindrome.js]: updated

File: alg-excer1-palindrome.js (hunk 4 above)
Comment (sureshrp) [reply alg-excer1-palindrome.js]: updated

File: alg-excer1-palindrome.js (hunk 1 above)
Comment (sureshrp) [reply alg-excer1-palindrome.js]: checking for numbers also

File: alg-excer1-palindrome.js (hunk 5 above)
Comment (sureshrp) [reply alg-excer1-palindrome.js]: updated, var

File: alg-excer1-palindrome.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
+function palindrome(input) {
+  var regExp = /[^A-za-z0-9]/g;
```
Comment (koding-101) [inline alg-excer1-palindrome.js]: Do not use anything you don't understand. How does regExp works? What is it doing? Do you know why it's inefficient? 

You should ALWAYS understand how something works or at least have any idea what it's doing before using it.

File: alg-excer1-palindrome.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+function checkPalindrome(palindrome) {
+
+  if (palindrome) {
+    console.log("Input value is palindrome")
+  } else {
+    console.log("Input value is not palindrome")
+  }
+}
+
+checkPalindrome(palindrome("S!A a!s"));
```
Comment (koding-101) [inline alg-excer1-palindrome.js]: you need to add more test cases.

File: alg-excer1-palindrome.js (hunk 6 above)
Comment (sureshrp) [reply alg-excer1-palindrome.js]: just used to replace non-alphanumeric characters
A-Z identifies anything that is not between A and Z
a-z identifies anything that is not between a and z
0-9 identifies anything that is not between 0 and 9

File: alg-excer1-palindrome.js (hunk 7 above)
Comment (sureshrp) [reply alg-excer1-palindrome.js]: added more tests

Comment (sureshrp) [reply alg-excer1-palindrome.js]: added more tests

**Comment by koding-101** [comment]
Merging - note again Regex is not needed for this assignment. Please try to solve it without regex when you get the chance.
